util/indexer/src/service.rs

```diff
@@ -433,6 +444,30 @@ impl IndexerHandle {
                     }
                 }
 
+                if let Some((data, mode)) = &filter_options.output_data {
+                    match mode {
+                        IndexerSearchMode::Prefix => {
+                            if !output_data.raw_data().starts_with(data) {
+                                return None;
+                            }
+                        }
+                        IndexerSearchMode::Exact => {
+                            if output_data.raw_data() != data {
+                                return None;
+                            }
+                        }
+                        IndexerSearchMode::Partial => {
+                            if !output_data
+                                .raw_data()
+                                .windows(data.len())
```

Comment:
An attacker may construct a query with zero data length which will cause the rpc server panic. Please add an unit test and check data length before call `windows` fn.

Reply:
OK

Reply:
Use `memchr::memmem::find` instead of `windows` for support of empty data.